x-pack/plugins/aiops/server/routes/explain_log_rate_spikes.ts

```diff
@@ -151,6 +156,105 @@ export const defineExplainLogRateSpikesRoute = (
           }
         }
 
+        if (changePoints?.length === 0) {
+          push(
+            updateLoadingStateAction({
+              ccsWarning: false,
+              loaded: 1,
+              loadingState: `Done.`,
+            })
+          );
+
+          end();
+          return;
+        }
+
+        const histogramFields: [NumericHistogramField] = [
+          { fieldName: request.body.timeFieldName, type: KBN_FIELD_TYPES.DATE },
+        ];
+
+        const [overallTimeSeries] = (await fetchHistogramsForFields(
+          client,
+          request.body.index,
+          { match_all: {} },
+          // fields
+          histogramFields,
+          // samplerShardSize
+          -1,
+          undefined
+        )) as [NumericChartData];
+
+        // time series filtered by fields
+        if (changePoints) {
+          await asyncForEach(changePoints, async (cp, index) => {
+            if (changePoints) {
+              const histogramQuery = {
+                bool: {
+                  filter: [
+                    {
+                      term: { [cp.fieldName]: cp.fieldValue },
+                    },
+                  ],
+                },
+              };
+
+              const [cpTimeSeries] = (await fetchHistogramsForFields(
+                client,
+                request.body.index,
+                histogramQuery,
+                // fields
+                [
+                  {
+                    fieldName: request.body.timeFieldName,
+                    type: KBN_FIELD_TYPES.DATE,
+                    interval: overallTimeSeries.interval,
+                    min: overallTimeSeries.stats[0],
+                    max: overallTimeSeries.stats[1],
+                  },
+                ],
+                // samplerShardSize
+                -1,
+                undefined
+              )) as [NumericChartData];
+
+              const histogram =
+                overallTimeSeries.data.map((o, i) => {
+                  const current = cpTimeSeries.data.find(
+                    (d1) => d1.key_as_string === o.key_as_string
+                  ) ?? {
+                    doc_count: 0,
+                  };
+                  return {
+                    key: o.key,
+                    key_as_string: o.key_as_string ?? '',
+                    doc_count_change_point: current.doc_count,
+                    doc_count_overall: Math.max(0, o.doc_count - current.doc_count),
+                  };
+                }) ?? [];
+
+              const { fieldName, fieldValue } = cp;
+
+              loaded += (1 / changePoints.length) * PROGRESS_STEP_HISTOGRAMS;
+              push(
+                updateLoadingStateAction({
+                  ccsWarning: false,
+                  loaded,
+                  loadingState: `Loading histogram data.`,
```

Comment:
Is this displayed in the UI?

Reply:
Added an item to the meta issue to translate those messages, there are more from previous PRs too. https://github.com/elastic/kibana/issues/136265